token.go

```diff
@@ -214,16 +214,32 @@ type TSTInfo struct {
 	Extensions     []pkix.Extension `asn1:"optional,tag:1"`
 }
 
-// Validate validates tst and returns the GenTime and Accuracy.
+// Validate validates tst and returns the Timestamp on success.
 // tst MUST be valid and the time stamped datum MUST match message.
-func (tst *TSTInfo) Validate(message []byte) (time.Time, time.Duration, error) {
+func (tst *TSTInfo) Validate(message []byte) (*Timestamp, error) {
 	if err := tst.validate(message); err != nil {
-		return time.Time{}, 0, err
+		return nil, err
 	}
-	accuracy := time.Duration(tst.Accuracy.Seconds)*time.Second +
-		time.Duration(tst.Accuracy.Milliseconds)*time.Millisecond +
-		time.Duration(tst.Accuracy.Microseconds)*time.Microsecond
-	return tst.GenTime, accuracy, nil
+
+	var accuracy time.Duration
+	// References:
+	// https://datatracker.ietf.org/doc/html/rfc3628#section-5.1
+	// https://github.com/notaryproject/specifications/blob/main/specs/trust-store-trust-policy.md#steps
+	if tst.Accuracy.Seconds == 0 &&
+		tst.Accuracy.Microseconds == 0 &&
+		tst.Accuracy.Milliseconds == 0 &&
+		oid.BaselineTimestampPolicy.Equal(tst.Policy) {
+		accuracy = 1 * time.Second
+	} else {
+		accuracy = time.Duration(tst.Accuracy.Seconds)*time.Second +
+			time.Duration(tst.Accuracy.Milliseconds)*time.Millisecond +
+			time.Duration(tst.Accuracy.Microseconds)*time.Microsecond
+	}
+
```

Comment:
@priteshbandi The implementation is based on RFC 3161 and Notary Project's spec:
https://github.com/notaryproject/specifications/blob/main/specs/trust-store-trust-policy.md
```
Retrieve the timestamp's accuracy. If the accuracy is explicitly specified in TSTInfo.accuracy, use that value. If the accuracy is not explicitly specified and TSTInfo.policy is the baseline time-stamp policy([RFC-3628](https://tools.ietf.org/html/rfc3628#section-5.2)), use accuracy of 1 second. Otherwise, use an accuracy of 0.
```
`use accuracy of 1 second` is reflected at line 232. In addition, we also take `Milliseconds` and `Microsecond` when accuracy is specified by TSA.